astropy/io/fits/hdu/hdulist.py

```diff
@@ -1518,14 +1518,28 @@ def _flush_resize(self):
                 # lead to odd behavior in practice.  Better to just not keep
                 # references to data from files that had to be resized upon
                 # flushing (on Windows--again, this is no problem on Linux).
-                # Note that this hack is only possible on numpy 1.x:
-                # in 2.x, we cannot write directly to the data attribute
                 for idx, mmap, arr in mmaps:
-                    if mmap is not None:
+                    if mmap is None:
+                        continue
+                    if NUMPY_LT_2_0:
+                        # Note that this hack is only possible on numpy 1.x:
+                        # in 2.x, we cannot write directly to the data attribute
                         # https://github.com/numpy/numpy/issues/8628
                         with warnings.catch_warnings():
                             warnings.simplefilter("ignore", category=DeprecationWarning)
                             arr.data = self[idx].data.data
+                    elif sys.getrefcount(arr) > 2:
+                        # 2 is the minimum number of references to this object
+                        # counting `arr` and a reference as an argument to getrefcount(),
+                        # see  https://docs.python.org/3/library/sys.html#sys.getrefcount
+                        warnings.warn(
+                            f"Memory map object {arr} with id={id(arr)} was closed but "
```

Comment:
printing `arr` doesn't seem a good idea since this is the "old" arr and will result in a segfault I think...

Reply:
🤦🏻‍♂️ Of course. Thanks !